File: SlidingWindow.js
const getMaxSumOfFiveContiguousElements = (arr) => {
  let maxSum = -Infinity;
  let currSum;

  for (let i = 0; i <= arr.length - 5; i++) {
    currSum = 0;

    for (let j = i; j <= i + 5; j++) {
      currSum += arr[j];
    }

    maxSum = Math.max(maxSum, currSum);
  }

  return maxSum;
};

File: SlidingWindow.py
def maxSum(arr):
    
    n = len(arr)

    if n < k:

        print("Invalid")            #so far sliding window has been challenging
        return -1

    window_sum = sum(arr[:k])

    max_sum = window_sum

    for i in range(n - k):
        window_sum = window_sum - arr[i] + arr[i + k]
        max_sum = max(window_sum, max_sum)

    return max_sum
    
